File: rabbit_mq/src/main/java/com/rabbit/rabbit_mq/global/security/CustomAuthenticationSuccessHandler.java
package com.rabbit.rabbit_mq.global.security;

import com.rabbit.rabbit_mq.domain.member.service.AuthTokenService;
import com.rabbit.rabbit_mq.global.https.ReqData;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.SavedRequestAwareAuthenticationSuccessHandler;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import java.io.IOException;

@Component
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class CustomAuthenticationSuccessHandler extends SavedRequestAwareAuthenticationSuccessHandler {
    private final ReqData rq;
    private final AuthTokenService authTokenService;

    @Override
    public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws ServletException, IOException {
        String accessToken = authTokenService.genAccessToken(rq.getMember());
        String refreshToken = rq.getMember().getRefreshToken();

        String redirectUrlAfterSocialLogin = rq.getCookieValue("redirectUrlAfterSocialLogin", "");

        if (rq.isFrontUrl(redirectUrlAfterSocialLogin)) {
            rq.destroySession();
            rq.setCrossDomainCookie("accessToken", accessToken);
            rq.setCrossDomainCookie("refreshToken", refreshToken);
            rq.removeCookie("redirectUrlAfterSocialLogin");

            response.sendRedirect(redirectUrlAfterSocialLogin);
            return;
        } else {
            rq.destroySession();
            rq.setCrossDomainCookie("accessToken", accessToken);
            rq.setCrossDomainCookie("refreshToken", refreshToken);
        }

        super.onAuthenticationSuccess(request, response, authentication);
    }
}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/global/Security/CustomAuthenticationSuccessHandler.kt
package com.krstudy.kapi.global.Security

import com.krstudy.kapi.domain.member.service.AuthTokenService
import com.krstudy.kapi.domain.member.service.MemberService
import com.krstudy.kapi.global.https.ReqData
import jakarta.servlet.ServletException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.SavedRequestAwareAuthenticationSuccessHandler
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional
import java.io.IOException
@Component
@Transactional(readOnly = true)
class CustomAuthenticationSuccessHandler(
    private val memberService: MemberService,
    private val rq: ReqData,
    private val authTokenService: AuthTokenService
) : SavedRequestAwareAuthenticationSuccessHandler() {

    @Throws(ServletException::class, IOException::class)
    override fun onAuthenticationSuccess(request: HttpServletRequest, response: HttpServletResponse, authentication: Authentication) {
        try {
            val member = memberService.getMemberByAuthentication(authentication) ?: run {
                response.sendRedirect("/member/login?error=memberNotFound")
                return
            }

            val token = authTokenService.genAccessToken(member)
            response.addHeader("Authorization", "Bearer $token")

            val redirectUrl = rq.getCookieValue("redirectUrlAfterSocialLogin", "/")
            response.sendRedirect(redirectUrl)
        } catch (e: Exception) {
            // 상세한 에러 로깅
            e.printStackTrace()
            logger.error("Authentication failed: ${e.message}", e)
            response.sendRedirect("/member/login?error=authenticationFailed")
        }
    }
}